collect.py

```diff
@@ -71,43 +75,67 @@ def store_results(temppath):
     print('Results stored in %s.' % fname)
 
 
+def parse_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        '-m', '--model',
+        help='Model to operate in. Accepts <controller name>:<model name>')
+    return parser.parse_args()
+
+
 def main():
+    options = parse_args()
+    model = options.model
+    if model:
+        if ':' in options.model and len(options.model.split(':')) == 2:
+            model = "-m {}".format(options.model)
+
     tempdir = tempfile.TemporaryDirectory()
     temppath = os.path.join(tempdir.name, 'results')
     os.makedirs(temppath)
 
     print('Getting juju status...')
     try:
-        raw_status = check_output('juju status --format json'.split())
+        raw_status = check_output(
+            'juju status {} --format json'.format(model).split())
         status = json.loads(raw_status.decode())
     except:
         print('Error getting juju status. Aborting.')
         return
 
-    debug_action(temppath, status, 'kubernetes-master')
-    debug_action(temppath, status, 'kubernetes-worker')
-    debug_action(temppath, status, 'etcd')
+    debug_action(temppath, status, model, 'kubernetes-master')
+    debug_action(temppath, status, model, 'kubernetes-worker')
+    debug_action(temppath, status, model, 'etcd')
     # FIXME: no debug action on kubeapi-load-balancer, easyrsa, flannel
 
-    command(temppath, 'status', 'juju status --format yaml')
-    command(temppath, 'debug-log', 'juju debug-log --replay')
-    command(temppath, 'model-config', 'juju model-config')
-    command(temppath, 'controller-debug-log', 'juju debug-log -m controller --replay')
-    command(temppath, 'storage', 'juju storage --format yaml')
-    command(temppath, 'storage-pools', 'juju storage-pools --format yaml')
-    command(temppath, 'kubernetes-master-config', 'juju config kubernetes-master --format yaml')
-    command(temppath, 'kubernetes-worker-config', 'juju config kubernetes-worker --format yaml')
-    command(temppath, 'kubeapi-load-balancer-config', 'juju config kubeapi-load-balancer --format yaml')
-    command(temppath, 'etcd-config', 'juju config etcd --format yaml')
-    command(temppath, 'easyrsa-config', 'juju config easyrsa --format yaml')
-    command(temppath, 'flannel-config', 'juju config flannel --format yaml')
+    command(temppath, 'status', 'juju status {} --format yaml'.format(model))
+    command(temppath, 'debug-log', 'juju debug-log {} --replay'.format(model))
+    command(temppath, 'model-config', 'juju model-config {}'.format(model))
+    command(temppath, 'controller-debug-log',
+            'juju debug-log -m controller --replay')
```

Comment:
OK, so the controller-debug-log is a little tricky. We need to get the model named `controller` from the controller that we're running against.

If I run `collect.py` with no model, or run `collect.py --model my-model` to select a model on the current controller, then we're good: `juju debug-log -m controller` is correct.

If I run `collect.py --model my-controller:my-model`, then we're operating on a controller named `my-controller`, so we need this command to be `juju debug-log -m my-controller:controller` instead.